Tools/clinic/clinic.py

```diff
@@ -4743,23 +4744,21 @@ def state_modulename_name(self, line: str | None) -> None:
         return_converter = None
         if returns:
             ast_input = f"def x() -> {returns}: pass"
-            module = None
             try:
-                module = ast.parse(ast_input)
+                module_node = ast.parse(ast_input)
             except SyntaxError:
-                pass
-            if not module:
-                fail("Badly-formed annotation for " + full_name + ": " + returns)
+                fail(f"Badly formed annotation for {full_name}: {returns!r}")
```

Comment:
Note: this error message is now slightly different to how it is on `main`. When I added a test, it became clear that we should use the `repr()` of `returns` in the error message here, rather than the `str()`. It makes it much easier to understand the error message, in my opinion.

(I also removed a redundant hyphen.)